jscomp/syntax/src/jsx_v4.ml

```diff
@@ -1299,12 +1296,7 @@ let transformSignatureItem ~config item =
       in
       let propsRecordType =
         makePropsRecordTypeSig ~coreTypeOfAttr ~typVarsOfCoreType "props"
-          psig_loc
-          ((* If there is Nolabel arg, regard the type as ref in forwardRef *)
-           (if !hasForwardRef then
-              [(true, "ref", [], Location.none, refTypeVar Location.none)]
-            else [])
-          @ namedTypeList)
+          psig_loc namedTypeList
```

Comment:
It looks like I implemented unnecessary handling in the initial version of JSX4. This handling is unnecessary because we typically use the labeled `~ref` in the interface of React.forwardRef implementations. Please see the tests I've added. https://github.com/rescript-lang/rescript-compiler/pull/6718/commits/34d17b77aa4ed0c46ac24bc627f233098c49e2f2